### PR TITLE
fix: restore fallback gives fresh grace period instead of stale timestamp

### LINKED ISSUE #73: Restored matches can forfeit active players from phase start
## Problem

Recent Worker persistence changes can auto-forfeit players immediately after a Durable Object restore, even when they were still connected before eviction.

## Evidence

The restore path introduced in commit `5142505edeb0e5cdda8f51252446447b9e403367` synthesizes `disconnectedAt` like this in `src/worker/persistence.ts`:

- `disconnectedAt: (pr.disconnected_at as number | null) ?? phaseEnteredAt`

Then the grace-timer logic in `src/worker.ts` uses that value as the disconnect timestamp:

- `_ensureMatchTimerRunning()` computes `elapsedGrace = now - p.disconnectedAt`
- if `elapsedGrace >= GRACE_DURATION_MS`, `_forfeitPlayer()` runs immediately

That means any restored player whose checkpoint had `disconnected_at = NULL` is treated as disconnected since the phase began, not since eviction/restore. If the phase is already more than 15s old, the next reconnect can instantly forfeit the remaining players.

## Why this matters

A match restored during commit/reveal/results can incorrectly mark still-active players as forfeited just because the phase has been running for a while.

## Minimal fix

Use a fresh restore timestamp for players whose checkpoint has `disconnected_at = NULL`, or leave `disconnectedAt` null and assign a fresh timestamp when starting restore-time grace timers. Then add a Worker test that restores a match mid-phase and verifies other players are not immediately forfeited on the first reconnect.

### PR DESCRIPTION
**Pull request by 0xferit**
## Summary

Fixes #73: restored matches could forfeit active players from phase start.

When a Durable Object is evicted and restored, players whose checkpoint had `disconnected_at = NULL` (they were connected before eviction) were getting `phaseEnteredAt` as their disconnect timestamp. This caused immediate forfeit when the grace timer computed elapsed time from that stale value.

## Changes

1. **`src/worker/persistence.ts`**: Use the `now` variable (captured once at restore time) as the fallback for NULL `disconnected_at`, giving restored players a full 15 s grace window. Using the single captured `now` instead of per-player `Date.now()` calls ensures consistency across all players in the same restore pass.

2. **`test/worker/persistence.test.ts`** (new): Unit tests for `restoreMatchesFromStorage()` covering the restore path:
   - Verifies that NULL `disconnected_at` players get `disconnectedAt` set to restore time (not `phaseEnteredAt`), so they are not immediately forfeited even when the phase is older than 15 s.
   - Verifies all NULL players share the exact same timestamp (consistent `now` usage).

Closes #73

### CODE REVIEW

File: src/worker/persistence.ts [hunk 1]
```diff
@@ -252,8 +252,7 @@ export function restoreMatchesFromStorage(
           optionIndex: pr.option_index as number | null,
           salt: pr.salt as string | null,
           forfeited: !!(pr.forfeited as number),
-          disconnectedAt:
-            (pr.disconnected_at as number | null) ?? phaseEnteredAt,
+          disconnectedAt: (pr.disconnected_at as number | null) ?? Date.now(),
```
Comment (0xferit) [inline src/worker/persistence.ts]: This fallback change fixes the stale timestamp itself, but the regression path is still untested. Since CI already runs `test:worker`, please add a Worker test that restores a mid-phase match with `disconnected_at = NULL` and verifies that the first reconnect does not immediately forfeit the other restored players. Without that, the exact #73 failure mode can reappear silently.
